docs: include rpk install instructions (#705)
redpanda-data/documentation: docs/quickstart/rpk-install.mdx

---
title: Installing rpk
---

import Tabs from "@theme/Tabs";
import TabItem from "@theme/TabItem";

Redpanda Keeper (rpk) is a single binary application written in [Go](https://go.dev/) to interact with your Redpanda clusters. It's designed to be an easy-to-use CLI for developers who need to build, maintain, and debug a production streaming system using Redpanda. 

> `rpk` is installed with the Redpanda broker, which is contained in all Redpanda packages. It also provides autocompletion, which helps you interact with easily using the tab button. For details, see [rpk documentation](../../reference/rpk/rpk-generate/rpk-generate-shell-completion).

Reviewer: I think "with the Redpanda broker" may be a bit odd -- I'd change it to (and merge the rest of the sentence with)" `with any Redpanda package`.

`It also` seems like an odd way to me to start a sentence, and this is a random spot to mention autocompletion. Most people forget about the autocompletion though, so it's good to have here -- the sentence just feels a bit awkward. Unfortunately, I don't have a good alternative.

Reviewer 2: > `It also` seems like an odd way to me to start a sentence

Indeed. Can we mention it at the end of the page simply like `To enable autocompletion, see <link>`?

Author: @daisukebe isn't autocompletion enabled by default?

Author: ```suggestion
`rpk` is bundled with Redpanda, so when you install Redpanda it also installs `rpk`. 

`rpk` features autocompletion, which helps you interact with easily using the tab button. For details, refer to [rpk documentation](../../reference/rpk/rpk-generate/rpk-generate-shell-completion).
```

Reviewer 2: > @daisukebe isn't autocompletion enabled by default?

No, users need to configure it manually.

Author: I addressed this with @AnneLeeperRP 's suggestion. Please reopen the conversation if you guys feel that we need to change the wording.

To install `rpk` bundled with Redpanda, refer to the [60-Second guides](../).

You can also install the `rpk` binary independently. 

<Tabs>
  <TabItem value="ubuntu" label="Linux" default>

Download the `rpk` archive for Linux with:

```bash
curl -LO https://github.com/redpanda-data/redpanda/releases/latest/download/rpk-linux-amd64.zip
```

Unpack the archive to `/usr/local/bin` or any location in your `$PATH` with:

```bash

> unzip rpk-linux-amd64.zip -d /usr/local/bin/

Reviewer: We may want to recommend some other path -- /usr/local/bin is not user writable, as much as the name implies. @andrewhsu recently recommended $HOME/.local/rpk/, but we'd also need to add an instruction to add this path you the user's $PATH.

Reviewer 2: Why don't we follow the way the kubectl doc does? https://kubernetes.io/docs/tasks/tools/install-kubectl-linux/

`Install kubectl binary with curl on Linux` --> `3. Install kubectl` --> `Note`

<img width="589" alt="image" src="https://user-images.githubusercontent.com/350206/195970606-0a55a029-edaa-4ac9-9b15-302758a210e8.png">

Reviewer: Let's recommend:
"Ensure your $PATH includes `~/.local/bin`, and then
```
mkdir -p ~/.local/bin
unzip rpk-linux-amd64.zip -d ~/.local/bin/
```
(perhaps worded better / formatting cleaned up)

Author: I've split the commands. Implemented as: 

Ensure your `$PATH` includes `~/.local/bin`:

```bash
mkdir -p ~/.local/bin
```

Execute the following command:

```bash
unzip rpk-linux-amd64.zip -d ~/.local/bin/
```

Author: WDYT?

```

  </TabItem>
  <TabItem value="macos" label="macOS" default>

Download the `rpk` archive for macOS with:

```bash

> curl -LO https://github.com/redpanda-data/redpanda/releases/latest/download/rpk-darwin-amd64.zip

Reviewer: This needs to be arm64 I think

Reviewer 2: kubectl doc mentions both amd64 and arm64, https://kubernetes.io/docs/tasks/tools/install-kubectl-macos/#install-kubectl-binary-with-curl-on-macos. Cockroach's ccloud doc does too https://www.cockroachlabs.com/docs/cockroachcloud/ccloud-get-started.html.

Author: I've added a tab for  Intel macOS (with amd64) and Apple Silicon(with arm64). What do you guys think?

```

Unpack the archive to `/usr/local/bin` or any location in your `$PATH` with:

```bash

> unzip rpk-darwin-amd64.zip -d /usr/local/bin/

Reviewer: Same path comment -- and it gets even weirder on mac, I think.

Reviewer 2: Seems like `/usr/local/bin` is fine on Mac as `/usr/libexec/path_helper` does add `/usr/local/bin` to $PATH when a shell launches. Confirmed it on my personal Mac too (Apple M1 Pro, macOS Monterey).

ref. https://unix.stackexchange.com/questions/356688/what-is-the-difference-between-echo-path-and-etc-paths

Author: @twmb resolution for this? I assume we can go forward with this.

Reviewer: Same comment as above

---
title: Installing rpk
---

import Tabs from "@theme/Tabs";
import TabItem from "@theme/TabItem";

Redpanda Keeper (rpk) is a single binary application written in [Go](https://go.dev/) to interact with your Redpanda clusters. 

It’s designed to be a Swiss Army Knife CLI application with simple commands that make it easy to use. It is handy not only for developers who want to evaluate Redpanda on their local machine, but also for developers who want to build, maintain, and debug a production streaming system using Redpanda.

> As mentioned above, `rpk` is a single binary application. 

Reviewer 2: Not sure this introduction is required in our doc. Could you advise, @Feediver1?

Reviewer 3: I think we can make this less "sales-y":

As mentioned previously, `rpk` is a single binary application, and designed to be an easy-to-use CLI for developers  who need to build, maintain, and debug a production streaming system using Redpanda.

Reviewer 3: ^^ @Deflaimun

Author: Resolved with @Feediver1 review.

> It’s designed to be a Swiss Army Knife CLI application with simple commands that make it easy to use. It is handy not only for developers who want to evaluate Redpanda on their local machine, but also for developers who want to build, maintain, and debug a production streaming system using Redpanda.
> 
> As mentioned above, `rpk` is a single binary application. 

Reviewer 3: ```suggestion
As mentioned previously, `rpk` is a single binary application, and designed to be an easy-to-use CLI for developers who need to build, maintain, and debug a production streaming system using Redpanda. 
```

> It is installed with the Redpanda broker, which is contained in all Redpanda packages. It also provides autocompletion, which helps you interact with rpk easily using the tab button. Please review the [rpk documentation](../../reference/rpk/) for more information.

Reviewer 2: Regarding autocompletion, I think we should link to https://docs.redpanda.com/docs/reference/rpk/rpk-generate/rpk-generate-shell-completion

Author: Good idea. Thanks !

> It is installed with the Redpanda broker, which is contained in all Redpanda packages. It also provides autocompletion, which helps you interact with rpk easily using the tab button. Please review the [rpk documentation](../../reference/rpk/) for more information.

Reviewer 3: ```suggestion
It is installed with the Redpanda broker, which is contained in all Redpanda packages. It also provides autocompletion, which helps you interact with `rpk` easily using the tab button. For details, see [rpk documentation](../../reference/rpk/).
```

> To install `rpk` bundled with Redpanda, check our [60-Second guides](../).

Reviewer 3: ```suggestion
To install `rpk` bundled with Redpanda, refer to the [60-Second guides](../).
```

> It's also possible to install the `rpk` binary independently. 

Reviewer 3: ```suggestion
You can also install the `rpk` binary independently. 
```

<Tabs>
<TabItem value="ubuntu" label="Ubuntu" default>

> Download the `rpk` archive for Ubuntu with:

Reviewer 2: Not `Ubuntu` specific. This is for any Linux distributions. @twmb , correct?

Reviewer 4: Yes, thanks @daisukebe

Author: Nice.

```bash
curl -LO https://github.com/redpanda-data/redpanda/releases/latest/download/rpk-linux-amd64.zip
```

Unpack the archive to `/usr/local/bin` or any location in your `$PATH` with:

```bash
unzip rpk-linux-amd64.zip
```

</TabItem>
<TabItem value="fedora" label="Fedora/RedHat" default>

Download the `rpk` archive for Fedora/RedHat with:

```bash

> curl -LO https://github.com/redpanda-data/redpanda/releases/latest/download/rpk-linux-amd64.zip

Author: @daisukebe can you review this link? I'm not sure what's the correct one for fedora/red hat

Reviewer 2: It just works in Fedora (or RH/CentOS) in the same way too. Confirmed below.

```
$ docker pull fedora:37
$ docker run --rm -ti fedora:37 bash
[root@9735e26ae11f /]# curl -LO https://github.com/redpanda-data/redpanda/releases/latest/download/rpk-linux-
[root@9735e26ae11f /]# sudo yum install unzip -y
[root@9735e26ae11f /]# unzip rpk-linux-amd64.zip
[root@9735e26ae11f /]# ./rpk --help
rpk is the Redpanda CLI & toolbox

Usage:
  rpk [command]

Available Commands:
  acl         Manage ACLs and SASL users
  cluster     Interact with a Redpanda cluster
  container   Manage a local container cluster
  debug       Debug the local Redpanda process
  generate    Generate a configuration template for related services
  group       Describe, list, and delete consumer groups and manage their offsets
  help        Help about any command
  iotune      Measure filesystem performance and create IO configuration file
  plugin      List, download, update, and remove rpk plugins
  redpanda    Interact with a local Redpanda process
  topic       Create, delete, produce to and consume from Redpanda topics
  version     Check the current version
  wasm        Deploy and remove inline WASM engine scripts

Flags:
  -h, --help      Help for rpk
  -v, --verbose   Enable verbose logging (default: false)

Use "rpk [command] --help" for more information about a command.
```

Author: Nice, thanks !

```

Unpack the archive to `/usr/local/bin` or any location in your `$PATH` with:

```bash
unzip rpk-linux-amd64.zip
```

> 

Reviewer 4: If we suggest they move the binary to a location in their `$PATH` can we give the full one-line command?

```bash
unzip rpk-linux-amd64.zip -d /usr/local/bin/
```

Author: Thanks @r-vasquez

</TabItem>

  <TabItem value="macos" label="macOS" default>

Download the `rpk` archive for macOS with:

```bash
curl -LO https://github.com/redpanda-data/redpanda/releases/latest/download/rpk-darwin-amd64.zip
```

Unpack the archive to `/usr/local/bin` or any location in your `$PATH` with:
```bash
unzip rpk-darwin-amd64.zip
```

  </TabItem>
</Tabs>

> Once it’s completed, run rpk version, which shows the rpk binary version, not the Redpanda server version.
> 

Reviewer 3: ```suggestion
After it’s completed, run `rpk version`, which shows the rpk binary version, not the Redpanda server version.

```

---
title: Installing rpk
---

import Tabs from "@theme/Tabs";
import TabItem from "@theme/TabItem";

Redpanda Keeper (rpk) is a single binary application that provides a way to interact with your Redpanda clusters from the command line. 

You can install `rpk` as part of a bundle that includes Redpanda (see the [60-Second Guides](../) for those instructions), or you can install the `rpk` binary separately. 

If you are preparing to deploy a BYOC cluster using the Redpanda Cloud UI, for example, you only need to install the `rpk` binary.

:::note
For a list of all `rpk` commands and their syntax, refer to [rpk documentation](../../reference/rpk).
:::

To install the rpk binary as a standalone application, follow the instructions for your operating system.

<Tabs>
  <TabItem value="ubuntu" label="Linux" default>

Download the `rpk` archive for Linux with:

```bash
curl -LO https://github.com/redpanda-data/redpanda/releases/latest/download/rpk-linux-amd64.zip
```

Ensure your `$PATH` includes `~/.local/bin`:

```bash
mkdir -p ~/.local/bin

> ```

Reviewer 5: The command that follows the instruction is just making a directory along with its parent directory; it's not ensuring that the directory is included in $PATH.

We need new text to introduce the `mkdir` command, and probably an additional step about updating $PATH with the new directory.

Execute the following command:

```bash
unzip rpk-linux-amd64.zip -d ~/.local/bin/

> ```

Reviewer 5: Instead of "Execute the following command:", can we describe what the command is doing? Would it be correct to say:
Unzip the `rpk` files to your `.local/bin/` directory:
(COMMAND)

Author: Good idea. Implemented.

  </TabItem>
  <TabItem value="macos" label="macOS" default>

> Download the `rpk` archive for macOS with:

Reviewer 5: Remove "with".

Author: done

```bash
curl -LO https://github.com/redpanda-data/redpanda/releases/latest/download/rpk-darwin-amd64.zip
```

Ensure your `$PATH` includes `~/.local/bin`:

```bash
mkdir -p ~/.local/bin

> ```

Reviewer 5: (Same comment as before for this text)
The command is not ensuring that $PATH includes the directory; it's just making the directory and the necessary parent directory.

We need to explain that the user has to create the directory, add it to $PATH, then unzip the files to that directory.

Author: Implemented as

> Ensure that you have the folder `~/.local/bin`:

> ```bash
> mkdir -p ~/.local/bin
> ```
> 
> Add it to your `$PATH`:
> 
> ```bash
> export PATH="~/.local/bin:$PATH"
> ```
> 
> Unzip the `rpk` files to your `.local/bin/` directory:
> 
> ```bash
> unzip rpk-linux-amd64.zip -d ~/.local/bin/
> ```

Author: @twmb could you approve the commands for accuracy?

> Execute the following command:

Reviewer 5: (General edit): Replace this with a description of what the command is doing.
Style Guide note: Use "run" instead of "execute" (except that the whole sentence will be replaced, so no need).

Author: done

<Tabs>

<TabItem value="intemac" label="Intel macOS" default>

```bash
unzip rpk-darwin-amd64.zip -d ~/.local/bin/
```
</TabItem>
<TabItem value="applesillicon" label="Apple Silicon" default>

```bash
unzip rpk-darwin-arm64.zip -d ~/.local/bin/
```
</TabItem>
</Tabs>

  </TabItem>
</Tabs>

> After it’s completed, run `rpk version`, which shows the rpk binary version, not the Redpanda server version.

Reviewer 5: What does "it" refer to? How about:
"After the files have been unzipped, run `rpk version` to display the `rpk` binary version."

We don't need to say that 'version' is not the server version; we can just say it's the binary version.

Author: done (good idea too! )

```bash
rpk version
```

The output looks like:

```bash
v22.1.2 (rev 449eef2)

> ```

Reviewer 5: Suggestion:
"The output displays the version number and the latest commit; for example:"
Also, how about using the latest GA version in the example? (That version will soon be out-of-date, but at least it's not as old as 22.1.2.)

Author: I would say that the latest hash commit is not as important. I would stick only to the version number.

And for the actual number, we can in the future have a injectable version that always gets the latest number. (this requires a bit of work, which this PR doesn't have at the moment.)

Author: ```suggestion
The output displays the version number:

```bash
v22.1.2 (rev 449eef2)
```
```

Author: Implemented as 

> The output displays the version number:
> 
> ```bash
> v22.1.2 (rev 449eef2)
> ```

---
title: Installing rpk
---

import Tabs from "@theme/Tabs";
import TabItem from "@theme/TabItem";

Redpanda Keeper (rpk) is a single binary application that provides a way to interact with your Redpanda clusters from the command line. 

> You can install `rpk` as part of a bundle that includes Redpanda (see the [60-Second Guides](../) for those instructions), or you can install the `rpk` binary separately. 

Reviewer: Should `for those instructions` just be `for instructions??

Author: done.

> You can install `rpk` as part of a bundle that includes Redpanda (see the [60-Second Guides](../) for those instructions), or you can install the `rpk` binary separately. 

Reviewer: Also, "that includes Redpanda" -- I think we prefer lowercase if we are referring to the binary itself

Author: Yes, that's true. done.

If you are preparing to deploy a BYOC cluster using the Redpanda Cloud UI, for example, you only need to install the `rpk` binary.

:::note
For a list of all `rpk` commands and their syntax, refer to [rpk documentation](../../reference/rpk).
:::

To install the rpk binary as a standalone application, follow the instructions for your operating system.

<Tabs>
  <TabItem value="ubuntu" label="Linux" default>

Download the `rpk` archive for Linux:

```bash
curl -LO https://github.com/redpanda-data/redpanda/releases/latest/download/rpk-linux-amd64.zip
```

Ensure that you have the folder `~/.local/bin`:

```bash
mkdir -p ~/.local/bin
```

Add it to your `$PATH`:

```bash
export PATH="~/.local/bin:$PATH"
```

> Unzip the `rpk` files to your `.local/bin/` directory:

Reviewer: Here and below, I'd prefix `./local/bin` with `~/`, to be explicit. `./` refers to the directory the person might currently be in, and every other usage of this has `~/`

Author: Yes. We must keep it accurate. good find.

```bash
unzip rpk-linux-amd64.zip -d ~/.local/bin/
```

  </TabItem>
  <TabItem value="macos" label="macOS" default>

Download the `rpk` archive for macOS:

```bash
curl -LO https://github.com/redpanda-data/redpanda/releases/latest/download/rpk-darwin-amd64.zip
```

Ensure that you have the folder `~/.local/bin`:

```bash
mkdir -p ~/.local/bin
```

Add it to your `$PATH`:

```bash
export PATH="~/.local/bin:$PATH"
```

Unzip the `rpk` files to your `.local/bin/` directory:
<Tabs>

> <TabItem value="intemac" label="Intel macOS" default>

Reviewer: `intemac` => `intelmac`?

Author: these are just html values, so it doesn't really matter. they just gotta be different.
Anyway, addressed.

```bash
unzip rpk-darwin-amd64.zip -d ~/.local/bin/
```
</TabItem>

> <TabItem value="applesillicon" label="Apple Silicon" default>

Reviewer: `sillicon` => `silicon`

Author: these are just html values, so it doesn't really matter. they just gotta be different.
Anyway, addressed.
